karapace/utils.py

```diff
@@ -48,3 +50,98 @@ def json_encode(obj, *, compact=True, sort_keys=True, binary=False):
         default=default_json_serialization
     )
     return res.encode("utf-8") if binary else res
+
+
+class Result:
+    def __init__(self, status, json_result, headers=None):
+        # We create both status and status_code so people can be agnostic on whichever to use
+        self.status_code = status
+        self.status = status
+        self.json_result = json_result
+        self.headers = headers if headers else {}
+
+    def json(self):
+        return self.json_result
+
+    def __repr__(self):
+        return "Result(status=%d, json_result=%r)" % (self.status_code, self.json_result)
+
+    def ok(self):
+        return self.status_code == 200
+
+
+class Client:
+    def __init__(self, server_uri=None, client=None):
+        self.server_uri = server_uri
+        self.session = requests.Session()
+        self.client = client
+
+    async def close(self):
+        try:
+            self.session.close()
+        except:  # pylint: disable=bare-except
+            pass
+        try:
+            await self.client.close()
+        except:  # pylint: disable=bare-except
+            pass
+
+    async def get(self, path, headers=None):
+        if not headers:
+            headers = {}
+        if self.client:
+            async with self.client.get(
+                path,
+                headers=headers,
+            ) as res:
+                json_result = await res.json()
+                return Result(res.status, json_result, headers=res.headers)
+        elif self.server_uri:
+            res = requests.get(os.path.join(self.server_uri, path), headers=headers)
```

Comment:
This might have be better done with either using `Path.as_posix()` or `urllib`.

Reply:
urllib is basically a PITA with trailing/ leading present / missing slashes, which made me use os.path, since i'm always merging a base url with other components without having to worry a leading slash will cancel all previous components